src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -679,7 +679,26 @@ private async Task<bool> EnumerateArtifact(IEnumeratedArtifact artifact, TContex
                 return false;
             }
 
-            string extension = Path.GetExtension(filePath);
+            string extension;
+            int lastDotIndex = filePath.LastIndexOf('.');
```

Comment:
>filePath [](http://example.com/codeflow?start=31&length=8)

you have added lots of code and you didn't add unit tests for this.

if you're going to inline Path.GetExtension, perhaps just grab the actual code (and get rid of the invalid path chars check).

can you please create a helper for this? also, did you look for other occurrences of this issue? i see other uses of Path.GetExtension in the code.

[https://referencesource.microsoft.com/#mscorlib/system/io/path.cs,f424e433705aeb09](https://referencesource.microsoft.com/#mscorlib/system/io/path.cs,f424e433705aeb09)

Reply:
None of the other uses of GetExtension looked like they were problematic to me and none of the telemetry points to any other code.

Reply:
I'm not yet convinced. The export rules and export config commands look like they could problematic to me as well. By inlining this change here and here only, you have left a non-obvious fix which compromises maintainability (i.e., it's very likely that if this problem is encountered again, a dev might not be aware of what you've done, add a duplicate helper or another inlined fix, etc.).

What I'd suggest is to create a static helper for this rewrite and please add unit tests. Then, add a new entry to IFileSystem for GetExtension and call this helper through IFileSystem everywhere in the code base.